src/UI/PersonNewsFeedBar.jsx

```diff
@@ -39,26 +65,42 @@ const PreferencesForm = () => {
     const sourcesString = activeSources.join(",");
 
     navigate(`/personalized-news/${sourcesString}`);
+    console.log("handle save");
+    toggleModal();
   };
 
+  if (!isOpen) return null;
+
   return (
     <div>
-      <h2>Set Your Preferences</h2>
-      {error && <div style={{ color: "red" }}>{error}</div>}
-      <div>
-        <h3>Sources:</h3>
-        {Object.keys(sources).map((source) => (
-          <label key={source}>
-            <input
-              type="checkbox"
-              checked={sources[source]}
-              onChange={() => handleSourceChange(source)}
-            />
-            {source}
-          </label>
-        ))}
-      </div>
-      <button onClick={handleSave}>Save Preferences</button>
+      <Dialog open={isOpen} PaperProps={{ ref: paperRef }}>
+        <DialogTitle>Set Your Preferences</DialogTitle>
+        {error && (
+          <DialogContentText style={{ color: "red" }}>
+            {error}
+          </DialogContentText>
+        )}
+        <DialogContent>
+          <DialogContentText>Sources:</DialogContentText>
```

Comment:
In order to prevent the user frustration, we can improve this section title by specifying more descriptive instruction, something like:

`Select at least one source:`

This way the user knows what to do before the error tells them